apps/evm/src/clients/api/queries/getPoolLiquidationIncentive/index.ts

```diff
@@ -1,23 +1,30 @@
 import BigNumber from 'bignumber.js';
 import { COMPOUND_MANTISSA } from 'constants/compoundMantissa';
-import type { IsolatedPoolComptroller, LegacyPoolComptroller } from 'libs/contracts';
+import { legacyPoolComptrollerAbi } from 'libs/contracts';
 import { convertPercentageFromSmartContract } from 'utilities';
+import type { Address, PublicClient } from 'viem';
 
 export interface GetPoolLiquidationIncentiveInput {
-  poolComptrollerContract: LegacyPoolComptroller | IsolatedPoolComptroller;
+  publicClient: PublicClient;
+  poolComptrollerContractAddress: Address;
 }
 
 export type GetPoolLiquidationIncentiveOutput = {
   liquidationIncentivePercentage: number;
 };
 
 export const getPoolLiquidationIncentive = async ({
-  poolComptrollerContract,
+  publicClient,
+  poolComptrollerContractAddress,
 }: GetPoolLiquidationIncentiveInput): Promise<GetPoolLiquidationIncentiveOutput> => {
-  const res = await poolComptrollerContract.liquidationIncentiveMantissa();
+  const liquidationIncentiveMantissa = await publicClient.readContract({
+    address: poolComptrollerContractAddress,
+    abi: legacyPoolComptrollerAbi,
+    functionName: 'liquidationIncentiveMantissa',
+  });
```

Comment:
This was compatible with both legacy and isolated pool, isn't it necessary anymore? Wouldn't we prefer to pass the isolated pools ABI?

Reply:
Because this part is shared between the legacy and the isolated `PoolComptroller` ABIs, then this function still supports legacy and isolated pools.
I've updated the code anyway to pass the isolated `PoolComptroller` ABI, since in similar situations we've tended to lean towards using the isolated pool implementation.